pango/src/language.rs

```diff
@@ -1,113 +1,25 @@
 // Take a look at the license at the top of the repository in the LICENSE file.
 
-use crate::Script;
-use glib::translate::*;
-use glib::GString;
+use std::str::FromStr;
 
-#[doc(alias = "PangoLanguage")]
-pub struct Language(*mut ffi::PangoLanguage);
+use crate::{Language, Script};
+use glib::translate::*;
 
 unsafe impl Send for Language {}
 unsafe impl Sync for Language {}
 
-#[doc(hidden)]
-impl<'a> ToGlibPtr<'a, *mut ffi::PangoLanguage> for &'a Language {
-    type Storage = &'a Language;
-
-    fn to_glib_none(&self) -> Stash<'a, *mut ffi::PangoLanguage, Self> {
-        Stash(self.0, *self)
-    }
-}
-
-#[doc(hidden)]
-impl<'a> ToGlibPtrMut<'a, *mut ffi::PangoLanguage> for Language {
-    type Storage = &'a mut Self;
-
-    #[inline]
-    fn to_glib_none_mut(&'a mut self) -> StashMut<'a, *mut ffi::PangoLanguage, Self> {
-        StashMut(self.0, self)
-    }
-}
-
-#[doc(hidden)]
-impl FromGlibPtrNone<*mut ffi::PangoLanguage> for Language {
-    unsafe fn from_glib_none(ptr: *mut ffi::PangoLanguage) -> Self {
-        assert!(!ptr.is_null());
-        Self(ptr)
-    }
-}
-
-#[doc(hidden)]
-impl FromGlibPtrFull<*mut ffi::PangoLanguage> for Language {
-    unsafe fn from_glib_full(ptr: *mut ffi::PangoLanguage) -> Self {
-        assert!(!ptr.is_null());
-        Self(ptr)
-    }
-}
-
-#[doc(hidden)]
-impl FromGlibPtrNone<*const ffi::PangoLanguage> for Language {
-    unsafe fn from_glib_none(ptr: *const ffi::PangoLanguage) -> Self {
-        assert!(!ptr.is_null());
-        Self(ptr as *mut _)
-    }
-}
-
-#[doc(hidden)]
-impl FromGlibPtrFull<*const ffi::PangoLanguage> for Language {
-    unsafe fn from_glib_full(ptr: *const ffi::PangoLanguage) -> Self {
-        assert!(!ptr.is_null());
-        Self(ptr as *mut _)
-    }
-}
-
-impl Default for Language {
```

Comment:
This got lost

Reply:
That seems like exactly the behaviour you'd want or not?

Reply:
And yes, someone might have used this :)

Reply:
> That seems like exactly the behaviour you'd want or not?

I thought Default is only meant for cases where you would get a new instance instead of the same one?

Reply:
`Language` is immutable or not? So that's the same effectively :)